xknx/devices/fan.py

```diff
@@ -101,6 +106,9 @@ def __str__(self) -> str:
 
     async def set_speed(self, speed: int) -> None:
         """Set the fan to a desginated speed."""
+
+        if self.mode == FanSpeedMode.Step and speed > 0:
+            speed = math.ceil(percentage_to_ranged_value(self._step_range, speed))
```

Comment:
I'd put math.ceil into percentage_to_ranged_value() so we woudn't have to think about it here (and in any other places it could be used)

Reply:
I would rather either keep the whole HA implementation unchanged or none of it. It's already tested, documented and we can drop in the whole file if there are any changes.  I know the logic is quite basic but it works ;)